### ISSUE TITLE
Problems with wrapper elements

### ISSUE THREAD

**Issue by wrygiel** [erasmus-without-paper/issues#18]
In Warsaw I have mentioned that wrapping `maxOccurs="unbounded"` elements in a wrapper element in considered a (relatively) bad practice. Here's one of the articles I found that describes this in more detail:

http://www.ibm.com/developerworks/library/x-contain/

Most relevant quotes:

> (...) The need for the books reference in programming often sets programmers up to expect to need such constructs in XML design. This is not the only carry-over that I am used to seeing. (...) In XML design, avoid constructions that do not correspond to any of the real-world phenomena you are modeling. You can immediately apply this test to Listing 2, as the element books doesn't really correspond to anything useful in the real-world phenomena being modeled (...)

And here are some problems caused by such design in the context of JAXB collections:

http://stackoverflow.com/questions/7338245/dealing-with-jaxb-collections
http://stackoverflow.com/questions/8728799/jaxb-generated-classes-ignore-intermediate-classes
http://stackoverflow.com/questions/2447091/how-generate-xmlelementwrapper-annotation-with-xjc-and-customized-binding

In short: There are tools to counteract such unnecessary wrappers in schema design, but they are usually used when XML Schema is already in place, and cannot be changed. In our case, we can still change the schema, so I think we should make use of this.

**Comment by wrygiel** [comment]
The article also explains that there are situations where the wrapper might be useful. I find it useful when I believe it's possible that other classes of wrapped elements are to be introduced in the future. For example, I thought about removing the [`<institutions-covered>` wrapper](https://github.com/erasmus-without-paper/ewp-specs-api-registry/blob/v1.0.2/catalogue-example.xml#L56) from the Registry API (and replacing it with `<hei-covered>` directly), but I concluded that it is somewhat possible that the Registry response might include other non-HEI institutions to our network in the future.

Anyway, I don't think there's anything to _do_ with this thread, so I'll close it now. I just wanted to expand on what I said in Warsaw. :)
